Add more modeling detail about Data Observability monitors (#34877)
DataDog/documentation: content/en/data_observability/quality_monitoring/guides/anomaly_detection.md

| Color | State | Description |
|-------|-------|-------------|
| Blue | Training | The monitor is learning baseline behavior. No alerts are triggered. |
| Green | Normal | The observed value is within expected bounds. |
| Red | Alerting | The observed value fell outside expected bounds. |

## Model states

The model often reuses prior predictions rather than generating new ones, for example when a value hasn't changed or when the monitor is in an alerting state. As a result, adding [annotations][2] is the primary way to get an alerting monitor back to a normal state and resume learning from new values.

## Metric-specific behavior

While the core model logic is shared across metric types, each type has additional rules:

| Metric type | Alert condition |
|-------------|----------------|
| Freshness | Time since last refresh is longer than expected based on historical update patterns. A small buffer is added to the upper bound to avoid alerting on minor delays. |
| Row count | Row count has not changed for longer than normal, indicating a stalled or broken pipeline. |
| Nullness, uniqueness | A jump to 0% or 100% when the metric has historically stayed away from those extremes. |

> | Custom SQL (Default model) | Value falls outside the expected range inferred from history. If the metric has never returned a negative value, the lower bound is constrained to 0. |

Reviewer: this table looks good, but i think we don't need both the table and the sections inside **Metric-specific behavior**. i'm more in favor of keeping just the sections and consolidating the info. what do you think?

The following seasonal patterns are detected:

| Pattern | Description |
|---------|-------------|
| Hour of day | Metrics that follow intraday patterns, such as higher row counts during business hours. |
| Hour of week | Metrics with consistent patterns across a full week at hourly granularity. |
| Day of week | Metrics that differ across days of the week, such as lower activity on Sundays. |
| Day of month | Metrics with recurring patterns tied to the calendar month, such as end-of-month spikes. |
| Weekday vs. weekend | Metrics with systematically different behavior on weekends versus weekdays. |

Not all seasonal patterns are available for all metric types. Additionally, the model requires multiple complete cycles of normal ("green") history before it can detect a given pattern.

## Trends

The model accounts for whether a metric is growing or shrinking over time. For a metric that consistently adds rows each week, the model adjusts expectations based on the direction and rate of change rather than treating growth as anomalous.

## Annotations

> Annotations let you provide feedback to the model when it misclassifies a point, either by alerting when it should not have or by missing an alert when it should have. Because data quality expectations are often business-specific, annotations are the primary way to tune the model to your team's needs over time.

Reviewer: ```suggestion
Annotations let you provide feedback to the model when it misclassifies a point, either by missing an alert or by generating a false alert. Because data quality expectations are often business-specific, annotations are the primary way to tune the model to your team's needs over time.
```
(we can play with the language, but i was trying to make this sentence a bit more concise)

---
title: Anomaly Detection for Data Observability Monitors
description: "Understand how the anomaly detection model works for Data Observability monitors, including training, model states, seasonality, and metric-specific behaviors."
further_reading:
  - link: '/data_observability/quality_monitoring/'
    tag: 'Documentation'
    text: 'Quality Monitoring'
  - link: '/monitors/types/data_observability/'
    tag: 'Documentation'
    text: 'Data Observability Monitor'
---

## Overview

> Data Observability monitors use an anomaly detection model tailored to common patterns in data quality metrics such as table freshness and row count. Unlike standard statistical anomaly detection, this model is designed for metrics that have irregular update patterns, sporadic large changes, and business-specific expectations of what is normal.

Reviewer 2: ```suggestion
Data Observability monitors use an anomaly detection model tailored to common patterns in data quality metrics such as table freshness and row count. This model is designed for metrics that have irregular update patterns, sporadic large changes, and business-specific expectations of what is normal.
```

The model learns from the metric's history to set expected bounds. When an observed value falls outside those bounds, the monitor triggers an alert.

## Training period

When you first create an anomaly detection monitor, it enters a **training period**. During training, the monitor collects historical values to learn the metric's baseline behavior. It does not trigger alerts during this period, and the monitor chart appears in blue.

The training period typically ends after the model has observed at least one weekday and one full weekend (Saturday and Sunday), since many data pipelines behave differently on weekends than on weekdays. Depending on how often the underlying metric is updated, training can take between **3 and 7 days**. Metrics that update infrequently (such as daily batch jobs) may require more observations before training completes.

After training completes, the monitor transitions to an active state and begins alerting. The chart shows green for normal values and red for outlier values.

> ## Model states

Reviewer 2: I think the typical reader will not really understand this section - it is dense and jargony. Consider simplifying & clarifying to the essential point: Users need to annotate to get monitors back to normal state after alert state. There is also some new auto-recovery logic so it isn't always required.

## Metric properties

The model analyzes up to 400 days of history to detect properties that inform how predictions are generated.

### Seasonality

Seasonality refers to regular, repeating patterns in a metric over time. The model detects and adjusts for the following seasonal patterns:

- **Hour of day**: Metrics that follow intraday patterns, such as higher row counts during business hours.
- **Hour of week**: Metrics with consistent patterns across a full week at hourly granularity.
- **Day of week**: Metrics that differ across days of the week (for example, lower activity on Sundays).
- **Day of month**: Metrics with recurring patterns tied to the calendar month, such as end-of-month spikes.
- **Weekday vs. weekend**: Metrics with systematically different behavior on weekends versus weekdays.

When seasonality is detected, the model shifts expected bounds to reflect the typical value at that point in the cycle rather than relying on a flat average.

### Stationarity

Stationarity describes whether a metric's average value is constant over time or trending in one direction.

> - **Stationary metrics** have a consistent average (for example, a table that is always refreshed with a similar row count).

Reviewer 2: ```suggestion
- **Stationary metrics** have a consistent average (for example, a row count for a table that occasionally adds or drops rows in equal measure).
```
